### ISSUE TITLE
Schema

### ISSUE THREAD

**Issue by melflynn** [tkla/DiscordClone#2]
- [x] Will users be able to upload their own avatars? If so, you probably don't need to store that in the users table. You'll be able to use ActiveStorage to handle this. 
- [x] I'm assuming that your "comments" will include original messages as well? 
- [ ] Users can only leave messages inside channels, correct? If so it may be a better plan of action to have a users_channels joins table since we can derive the server_id from the channel but not the other way around. Let me know if I'm not correctly understanding your approach here
- [x] What is your plan for voice? I think if you intend to build out voice channels that should be a separate MVP

**Comment by tkla** [comment]
- [x] Will users be able to upload their own avatars? If so, you probably don't need to store that in the users table. You'll be able to use ActiveStorage to handle this. 

That sounds very useful, I'll have to look more into that as I'm not familiar with ActiveStorage. I'll remove the avatar columns in the meantime. I added a active storage table to enable the association with files <-> users.

- [x ] I'm assuming that your "comments" will include original messages as well? 
  Added column called `original_body`

- [x ] Users can only leave messages inside channels, correct? If so it may be a better plan of action to have a users_channels joins table since we can derive the server_id from the channel but not the other way around. Let me know if I'm not correctly understanding your approach here

I think I'm misunderstanding, but yes users can only leave comments in channels. I'm confused about what you're suggesting here. 

`users_servers` exists as a join table to create a many to many relation between Users and Servers to establish the users in a server and the servers a user has joined.
 
- [x ] What is your plan for voice? I think if you intend to build out voice channels that should be a separate MVP
To be honest voice channels feels more like a stretch goal than an MVP.
